### PR TITLE
Adding ConfigClass for HTTPX

### PR DESCRIPTION
**Pull request by mattjacksoncello**
This PR is to implement a config class to allow more control over httpx for things like certificate verification.

I have also merged in the changes from #92 

Note: I have used Claude to help with the changes. I have reviewed everything but I am not the strongest coder so there may be mistakes that I have missed

### CODE REVIEW

File: src/onepasswordconnectsdk/__init__.py [hunk 1]
```diff
@@ -4,12 +4,5 @@
...
 from onepasswordconnectsdk.client import new_client_from_environment
-
-__all__ = [
-    "client",
-    "load",
-    "load_dict",
-    "models",
-    "new_client",
-    "new_client_from_environment"
-]
```
Comment (AndyTitu) [inline src/onepasswordconnectsdk/__init__.py:15]: Won't deleting this block avoid these components from being properly exported from this package?

File: src/tests/test_client_vaults.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 HOST = "https://mock_host"
 TOKEN = "jwt_token"
-SS_CLIENT = client.new_client(HOST, TOKEN)
-SS_CLIENT_ASYNC = client.new_client(HOST, TOKEN, True)
+# Create clients using new config pattern
+client_config = ClientConfig(url=HOST, token=TOKEN)
+SS_CLIENT = client.Client(client_config)
+
+async_config = AsyncClientConfig(url=HOST, token=TOKEN)
+SS_CLIENT_ASYNC = async_client.AsyncClient(async_config)
```
Comment (AndyTitu) [inline src/tests/test_client_vaults.py:15]: @mattjacksoncello could you

File: src/onepasswordconnectsdk/client.py [hunk 3]
```diff
@@ -381,34 +388,34 @@ def sanitize_for_serialization(self, obj):
         return self.serializer.sanitize_for_serialization(obj)
 
 
-def new_client(url: str, token: str, is_async: bool = False) -> Union[AsyncClient, Client]:
+def new_client(url: str, token: str, **kwargs) -> Client:
```
Comment (AndyTitu) [inline src/onepasswordconnectsdk/client.py:391]: This change is not backwards compatible unfortunately, and other existing user scripts will break if we ever released this and those users updated to this new version. To support this new separation we'd have to release a new major version of the SDK, namely 2.0, and I'm not sure it's worth it only to separate the client initialisation function. 

Don't get me wrong, I do agree that this would probably be a better pattern, it's just that I don't believe that this improvement alone is worth releasing a 2.0. Can we maybe revert this client separation and only keep the client config part of this PR?

**Comment by mattjacksoncello** [comment]
Thanks for the review Andy, will refactor the code

**Comment by mattjacksoncello** [comment]
@AndyTitu,

I've updated the PR to focus solely on the config class changes as requested. Here are the key updates:

1. Documentation:
   - Added a new section in USAGE.md documenting the primary use case for CA certificate verification with Connect server
   
2. Testing:
   - Created a new test file specifically for the config class implementation
   - Looking for feedback on test coverage and potential edge cases
   
3. Examples:
   - Added a new example demonstrating CA file usage in practice

Please review these changes and let me know if any adjustments are needed.

**Comment by mattjacksoncello** [comment]
Hi @AndyTitu 

Any updates on this?

Thanks

**Comment by mattjacksoncello** [comment]
Hi @AndyTitu 
When do you think you will have time to review this? I would like to use the production version if possible.
Thanks

**Comment by AndyTitu** [comment]
@mattjacksoncello sorry for missing notifications on this. I'm on it now.

File: src/onepasswordconnectsdk/config.py [hunk 4]
```diff
@@ -16,7 +19,48 @@
...
+    Inherits from httpx.BaseClient to support all httpx client options.
+    """
+    def __init__(self, cafile: Optional[str] = None, **kwargs):
+        """Initialize client configuration
+
+        Args:
+            cafile (Optional[str]): Path to CA certificate file for SSL verification
+            **kwargs: Additional httpx client options
+        """
+        self.cafile = cafile
```
Comment (AndyTitu) [inline src/onepasswordconnectsdk/config.py:33]:
```suggestion
        self.ca_file = ca_file
```

**Review by AndyTitu — APPROVED** [review]
Code looks good to me apart from some small suggestions, thank you for the contribution! ❤️

**Comment by AndyTitu** [comment]
@mattjacksoncello some tests are failing for the last changes.

**Comment by mattjacksoncello** [comment]
Hi Andy,
I have renamed all instances of cafile to ca_file. Test are now working
Cheers!

**Comment by mattjacksoncello** [comment]
@AndyTitu Could you have another look at this please

**Review by AndyTitu — APPROVED** [review]

**Comment by mattjacksoncello** [comment]
Hi @AndyTitu  
when will this be integrated into a release?
